create_commits.py

```diff
@@ -0,0 +1,347 @@
+
+    return file_changes
+
+
+def get_local_commits_not_on_remote(
+    local_branch_name: str, remote_branch_name: str
+) -> list[str]:
+    """
+    Get a list of commit hashes on the local branch that are not on the remote branch,
+    chronologically ordered from oldest to newest.
+
+    Args:
+        local_branch_name (str): The name of the local branch.
+        remote_branch_name (str): The name of the remote branch.
+
+    Returns:
+        list: A list of strings representing the commit hashes.
+    """
+    result = subprocess.run(
+        ["git", "rev-list", f"origin/{remote_branch_name}..{local_branch_name}"],
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    logging.info(
+        "Found %s commits on the local branch that are not on the remote branch.",
+        len(result.stdout.splitlines()),
+    )
+    logging.debug("Commits to be created on the remote branch:")
+
+    for commit_hash in result.stdout.splitlines():
+        logging.debug(commit_hash)
+
+    # reverse the list so that the oldest commit is first
+    return result.stdout.splitlines()[::-1]
+
+
+def create_commit_on_remote_branch(
+    *,
+    github_token: str,
+    repository_name_with_owner: str,
+    remote_branch_name: str,
+    expected_head_oid: str,
+    file_changes: FileChanges,
+    message: CommitMessage,
+) -> str:
+    """
+    Create a commit on a remote branch using the GitHub GraphQL API.
+
+    Args:
+        github_token (str): The GitHub personal access github_token.
+        repository_name_with_owner (str): The name of the repository with the owner.
+        remote_branch_name (str): The name of the branch.
+        expected_head_oid (str): The expected git commit oid at the head of the branch prior to the commit.
+        file_changes (dict): The file changes object.
+        message (str): The commit message.
+
+    Returns:
+        str: The commit oid of the created commit.
+    """
+
+    assert not remote_branch_name.startswith(
+        "origin/"
+    ), "Do not include 'origin/' in the remote branch name."
+
+    assert not remote_branch_name.startswith(
+        "refs/heads/"
+    ), "Do not include 'refs/heads/' in the remote branch name."
+
+    url = "https://api.github.com/graphql"
+    headers = {"Authorization": f"Bearer {github_token}"}
+
+    mutation = """
+    mutation ($input: CreateCommitOnBranchInput!) {
+      createCommitOnBranch(input: $input) {
+        commit {
+          oid
+        }
+      }
+    }
+    """
+
+    graphql_input = {
+        "branch": {
+            "repositoryNameWithOwner": repository_name_with_owner,
+            "branchName": remote_branch_name,
+        },
+        "expectedHeadOid": expected_head_oid,
+        "fileChanges": file_changes,
+        "message": message,
+    }
+
+    data = {"query": mutation, "variables": {"input": graphql_input}}
+    response = requests.post(url, headers=headers, json=data)
```

Comment:
omg nooooo way!!!!!! 😮

Reply:
discussed via slack - we unfortunately can't do this :( since each mutation needs the result of the second, to pass as the `expected_head_oid` parameter.